flight/Modules/Autotune/autotune.c

```diff
@@ -209,13 +215,38 @@ static void at_new_actuators(UAVObjEvent * ev, void *ctx, void *obj, int len) {
 
 			if (circ_queue_advance_write(at_queue) == 0) {
 				running = false;
+				first_cycle = false;
 			}
 		}
 
 		return;		// No data
 	}
 
-	running = true;
+	/* Simple state machine.
+	 * !running !first_cycle -> running first_cycle -> running !first_cycle
+	 */
+	if (!running || first_cycle) {
+		if (actuators->SystemIdentCycle == 0x0000) {
+			running = true;
+			first_cycle = ! first_cycle;
+		}
+	}
+
+#ifdef AUTOTUNE_AVERAGING_MODE
+	struct at_measurement *avg_point = &at_averages[actuators->SystemIdentCycle];
+
+	if (first_cycle) {
+		bzero(avg_point, sizeof(*avg_point));
```

Comment:
bzero probably not a good idea for new code. Deprecated, and removed in newer POSIX standards.